include/service/entities/site_service.js

```diff
@@ -123,6 +142,16 @@ module.exports = function SiteServiceModule(pb) {
     };
 
     /**
+     * Get all of the redirect_hosts in the database
+     * @method getRedirectHosts
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getRedirectHosts = function(cb) {
+      var dao = new pb.DAO();
+      dao.q(REDIRECT_HOSTS, { select: pb.DAO.SELECT_ALL, where: {} }, cb);
+    };
```

Comment:
You and I are on the same page but saying different things.

Let's say you have a site object:
```
{
  hostname: "test1.localhost:8080",
  prevHostNames: [
    "old.localhost:8080"
  ]
}
```

It gets passed to ```RequestHandler.loadSite(site);```

```
function(site) {
  RequestHandler.sites[site.hostname] = { active: site.active, uid: site.uid, displayName:  site.displayName, hostname: site.hostname };
  site.prevHostNames.forEach(function(oldHostname) {
    RequestHandler.siteRedirects[oldHostname] = site.hostname;
  }
}
```

You lookup for each request is constant time and you avoid the iteration.